templates/types/streaming/nextjs/app/components/ui/chat/chat-message/index.tsx

```diff
@@ -32,19 +27,13 @@ type ContentDisplayConfig = {
   component: JSX.Element | null;
 };
 
-function ChatMessageContent({
-  message,
-  isLoading,
+export function ChatMessageContent({
```

Comment:
all except `ChatTools` can be moved to chat-ui, then the code here would be as simple as:
```typescript
export function CustomChatMessageContent() {
  const { annotations } = useChatMessage();

  const toolData = getAnnotationData<ToolData>(
    annotations,
    MessageAnnotationType.TOOLS, // custom type defined in create-llama
  ); // should be undefined if there's no tool data in the annotations

  const content = toolData
    ? [
        {
          position: ContentPosition.AFTER_EVENTS, // constant from @llamaindex/chat-ui` that defines where the content should be displayed
          component: <ChatTools data={toolData[0]} />,
        },
      ]
    : [];

  return (
    <div className="flex-1 gap-4 flex flex-col">
      <ChatMessageContent content={content} /> // renders all the content as is
    </div>
  );
}
```